src/Humanizer.Tests.Shared/Localisation/fr/TimeToClockNotationTests.cs

```diff
@@ -0,0 +1,64 @@
+﻿#if NET6_0_OR_GREATER
+
+using System;
+using Xunit;
+using Humanizer.Localisation.TimeToClockNotation;
+using Humanizer;
+
+namespace Humanizer.Tests.Localisation.fr
+{
+    [UseCulture("fr")]
+    public class TimeToClockNotationTests
+    {
+        [Theory]
+        [InlineData(00, 00, "minuit")]
+        [InlineData(01, 00, "une heure")]
+        [InlineData(04, 00, "quatre heures")]
+        [InlineData(05, 01, "cinq heures une")]
+        [InlineData(06, 05, "six heures cinq")]
+        [InlineData(07, 10, "sept heures dix")]
+        [InlineData(08, 15, "huit heures et quart")]
+        [InlineData(09, 20, "neuf heures vingt")]
+        [InlineData(10, 25, "dix heures vingt-cinq")]
+        [InlineData(11, 30, "onze heures et demie")]
+        [InlineData(12, 00, "midi")]
+        [InlineData(15, 35, "quinze heures trente-cinq")]
+        [InlineData(16, 40, "dix-sept heures moins vingt")]
+        [InlineData(17, 45, "dix-huit heures moins le quart")]
+        [InlineData(18, 50, "dix-huit heures cinquante")]
+        [InlineData(19, 55, "dix-neuf heures cinquante-cinq")]
+        [InlineData(20, 59, "vingt heures cinquante-neuf")]
+        public void ConvertToClockNotationTimeOnlyString(int hours, int minutes, string expectedResult)
+        {
+            var actualResult = new TimeOnly(hours, minutes).ToClockNotation();
+            Assert.Equal(expectedResult, actualResult);
+        }
+
+        [Theory]
+        [InlineData(00, 00, "minuit")]
+        [InlineData(04, 00, "quatre heures")]
+        [InlineData(05, 01, "cinq heures")]
+        [InlineData(06, 05, "six heures cinq")]
+        [InlineData(07, 10, "sept heures dix")]
+        [InlineData(08, 15, "huit heures et quart")]
+        [InlineData(09, 20, "neuf heures vingt")]
+        [InlineData(10, 25, "dix heures vingt-cinq")]
+        [InlineData(11, 30, "onze heures et demie")]
+        [InlineData(12, 00, "midi")]
+        [InlineData(13, 23, "treize heures vingt-cinq")]
+        [InlineData(14, 32, "quatorze heures et demie")]
+        [InlineData(15, 35, "quinze heures trente-cinq")]
+        [InlineData(16, 40, "dix-sept heures moins vingt")]
+        [InlineData(17, 45, "dix-huit heures moins le quart")]
+        [InlineData(18, 50, "dix-huit heures cinquante")]
+        [InlineData(19, 55, "dix-neuf heures cinquante-cinq")]
```

Comment:
That's just it, we're in a grey area here. I too am a native French speaker and, though both forms are accepted, I would normally use the wording validated by this test. (But I'm not totally unbiased, am I? 😆)

Reply:
I dug a bit more, to find out in which context we should use either form. It seems that to express, say, `7:55 pm`
1. the **formal way** is `dix-neuf heures cinquante-cinq`,
1. the **informal way** would be `huit heures moins cinq`, but
1. you should never say `vingt heures moins cinq`

Having said that, which of 1 and 2 should we prefer here?

References:
- https://francais.lingolia.com/fr/vocabulaire/nombres-date-et-heure/lheure
- https://leflepourlescurieux.fr/dire-l-heure-maniere-formelle-et-maniere-informelle-grammaire-a1/

Reply:
Alright let's forget the third point. For the formal and the informal way, I don't have a strict opinion. Since we are trying to humanize the content, I would choose the informal way.

Reply:
The informal way like 2 doesn't specify if it's AM or PM. Personally I would choose the formal way.

Reply:
@p0utine true. Let's keep the formal way.